.github/actions/stable-changelog/index.js

```diff
@@ -0,0 +1,88 @@
+const core = require("@actions/core");
+const github = require("@actions/github");
+const fs = require("fs");
+const readline = require("readline");
+try {
+  const head_version = fs
+    .readFileSync(core.getInput("version-path"), "utf8")
+    .trim();
+  const changelog = fs.readFileSync(core.getInput("changelog-path"), "utf8");
+  const rl = readline.createInterface({
+    input: fs.createReadStream(core.getInput("changelog-path")),
+  });
+  const re = /\[\d+\.\d+\.\d+(-\d+)*?\]/;
+  const summary = {
+    Authors: [],
+    Added: [],
+    Changed: [],
+    Removed: [],
+    Fixed: [],
+    Tests: [],
+    Docs: [],
+    Operations: [],
+  };
+  let curline = 0;
+  let begin = Number.MAX_SAFE_INTEGER;
+  let end = Number.MAX_SAFE_INTEGER;
+  let section, i, last_release;
+  rl.on("line", (text) => {
+    if (text.match("\\b" + head_version + "\\b")) {
+      begin = curline++;
+      return;
+    }
+    if (curline < begin || curline > end) {
+      curline++;
+      return;
+    }
+    if (text.match(re)) {
+      end = curline++;
+      last_release = text.match(re)[0];
+      rl.close();
+      return;
+    }
+    console.log(text);
+    if (text.includes("### Authors")) section = "Authors";
+    else if (text.includes("### Added")) section = "Added";
+    else if (text.includes("### Changed")) section = "Changed";
+    else if (text.includes("### Removed")) section = "Removed";
+    else if (text.includes("### Fixed")) section = "Fixed";
+    else if (text.includes("### Tests")) section = "Tests";
+    else if (text.includes("### Docs")) section = "Docs";
+    else if (text.includes("### Operations")) section = "Operations";
+    else if (text.includes("##")) section = null;
+    else if (text.length > 2 && section === "Authors") {
+      i = 0;
+      while (text && i < summary.Authors.length) {
```

Comment:
Seems the while loop and keeping track of index could be made more concise
```js
const isAuthorAlreadyIncluded = summary.Authors.find(author => author.includes(text.split(/[<>]/)[1]))
if(!isAuthorAlreadyIncluded) 
  summary.Authors.push(text);
```

I'll leave this up to you if you find this more readable